packages/backend-core/src/cache/writethrough.ts

```diff
@@ -84,16 +84,16 @@ async function put(
   return { ok: true, id: output._id, rev: output._rev }
 }
 
-async function get(db: Database, id: string): Promise<any> {
+async function get<T extends Document>(db: Database, id: string): Promise<T> {
   const cache = await getCache()
   const cacheKey = makeCacheKey(db, id)
   let cacheItem: CacheItem = await cache.get(cacheKey)
   if (!cacheItem) {
-    const doc = await db.get(id)
+    const doc = await db.get<T>(id)
     cacheItem = makeCacheItem(doc)
     await cache.store(cacheKey, cacheItem)
   }
-  return cacheItem.doc
+  return cacheItem.doc as T
```

Comment:
Is this not typed by default?

Reply:
No it wasn't, but I've switched from casting it to making the typing part of the `CacheItem` - which is a bit better.